[quant][graphmode] Make `aten::relu` a general op

jerryzh168 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #35571 [quant][graphmode] Quantization support for at::sort
* #35558 [quant][graphmode][refactor] getGeneralTensorInputs(Node*) -> getPassThroughInputs(Value*)
* #35557 [quant][graphmode][fix] docs for InsertObservers
* #35509 [quant][graphmode] Quantization support for `quantized::add_scalar_relu` and `quantized::add_scalar_relu_out`
* #35385 [quant][graphmode] Remove unused patterns
* #35347 [quant][graphmode] Quantization support for `aten::conv3d`
* #35345 [quant][graphmode][fix] Add filter for `quantized::add`
* #35336 [quant][graphmode] qconfig_dict support None
* **#35420 [quant][graphmode] Make `aten::relu` a general op**
* #35335 [quant][graphmode] `prepare_script` takes original qconfig_dict

Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

jerryzh168 committed:
[quant][graphmode] Make `aten::relu` a general op

Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

z-a-f reviewed (approved):
Do you need to change any calls in the unittests?

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]

jerryzh168 committed:
Update on "[quant][graphmode] Make `aten::relu` a general op"


Summary:
This PR makes `aten::relu` a general op that doesn't require observation

This means we also need to change the logic to support skipping intermediate values because
this breaks `conv - relu` pattern if it is not followed by something that is quantizable
since `conv` is quantizable, but we decide to skip observing between conv and relu.

We changed the old `skip_values` to a new `delay_observation_map_` which records information that
allow us to delay the observation of certain values until later points. In the case of `conv - relu`
pattern, we delayed the observation of output of `conv` and observe the output of `relu` instead.

Test Plan:
python test/test_jit.py

Reviewers:
.

Subscribers:

Tasks:

Tags:

Differential Revision: [D20655309](https://our.internmc.facebook.com/intern/diff/D20655309)

[ghstack-poisoned]
